### ISSUE TITLE
deployed on vercel

### ISSUE THREAD

**Issue by vishalkadam7** [Lissy93/dashy#1465]
deployed on vercel

I get this error while trying to save config. is there any thing else that i need to configure in vercel to get the config saved

AxiosError: Request failed with
status code 405

_Originally posted by @vishalkadam7 in https://github.com/Lissy93/dashy/issues/1147#issuecomment-1933647816_

**Comment by vishalkadam7** [comment]
Can someone guide me please

**Comment by CrazyWolf13** [comment]
> ```
>           deployed on vercel
> ```
> 
> I get this error while trying to save config. is there any thing else that i need to configure in vercel to get the config saved
> 
> AxiosError: Request failed with status code 405
> 
> _Originally posted by @vishalkadam7 in [#1147 (comment)](https://github.com/Lissy93/dashy/issues/1147#issuecomment-1933647816)_

I sadly cannot really help you that much regarding the error. As I do not personally know vercel.

Please ensure you use this Template inside Vercel ```https://vercel.com/new/project?template=https://github.com/lissy93/dashy```

Like in the wiki: https://dashy.to/docs/deployment#vercel

Have you done a rebuild?

**Comment by vishalkadam7** [comment]
Everything is done properly I even get the dashboard but after editing the config don't save,

And I'm not even able to build from the source code it shows some compatibility issues

**Comment by CrazyWolf13** [comment]
Just spun up Vercel, can confirm this issue.

![image](https://github.com/Lissy93/dashy/assets/96661824/57e73810-ee43-4e34-bc79-aa794592b723)


Deploy errors:
A bit too many erros....

`````
<html>
<body>
<!--StartFragment-->
warning " > @babel/preset-env@7.18.2" has unmet peer dependency "@babel/core@^7.0.0-0".
--
20:14:30.513 | warning "@babel/preset-env > @babel/helper-compilation-targets@7.18.2" has unmet peer dependency "@babel/core@^7.0.0".
20:14:30.513 | warning "@babel/preset-env > @babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression@7.17.12" has unmet peer dependency "@babel/core@^7.0.0".
20:14:30.514 | warning "@babel/preset-env > @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@7.17.12" has unmet peer dependency "@babel/core@^7.13.0".
20:14:30.516 | warning "@babel/preset-env > @babel/plugin-proposal-async-generator-functions@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.516 | warning "@babel/preset-env > @babel/plugin-proposal-class-properties@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.516 | warning "@babel/preset-env > @babel/plugin-proposal-class-static-block@7.18.0" has unmet peer dependency "@babel/core@^7.12.0".
20:14:30.516 | warning "@babel/preset-env > @babel/plugin-proposal-dynamic-import@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.517 | warning "@babel/preset-env > @babel/plugin-proposal-export-namespace-from@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.517 | warning "@babel/preset-env > @babel/plugin-proposal-json-strings@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.517 | warning "@babel/preset-env > @babel/plugin-proposal-logical-assignment-operators@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.517 | warning "@babel/preset-env > @babel/plugin-proposal-nullish-coalescing-operator@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.518 | warning "@babel/preset-env > @babel/plugin-proposal-numeric-separator@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.518 | warning "@babel/preset-env > @babel/plugin-proposal-object-rest-spread@7.18.0" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.518 | warning "@babel/preset-env > @babel/plugin-proposal-optional-catch-binding@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.518 | warning "@babel/preset-env > @babel/plugin-proposal-optional-chaining@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.518 | warning "@babel/preset-env > @babel/plugin-proposal-private-methods@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.519 | warning "@babel/preset-env > @babel/plugin-proposal-private-property-in-object@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.519 | warning "@babel/preset-env > @babel/plugin-proposal-unicode-property-regex@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.519 | warning "@babel/preset-env > @babel/plugin-syntax-async-generators@7.8.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.519 | warning "@babel/preset-env > @babel/plugin-syntax-class-properties@7.12.13" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.520 | warning "@babel/preset-env > @babel/plugin-syntax-class-static-block@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.520 | warning "@babel/preset-env > @babel/plugin-syntax-dynamic-import@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.520 | warning "@babel/preset-env > @babel/plugin-syntax-export-namespace-from@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.520 | warning "@babel/preset-env > @babel/plugin-syntax-import-assertions@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.520 | warning "@babel/preset-env > @babel/plugin-syntax-json-strings@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.521 | warning "@babel/preset-env > @babel/plugin-syntax-logical-assignment-operators@7.10.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.521 | warning "@babel/preset-env > @babel/plugin-syntax-nullish-coalescing-operator@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.521 | warning "@babel/preset-env > @babel/plugin-syntax-numeric-separator@7.10.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.521 | warning "@babel/preset-env > @babel/plugin-syntax-object-rest-spread@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.521 | warning "@babel/preset-env > @babel/plugin-syntax-optional-catch-binding@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-syntax-optional-chaining@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-syntax-private-property-in-object@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-syntax-top-level-await@7.14.5" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-transform-arrow-functions@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-transform-async-to-generator@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.522 | warning "@babel/preset-env > @babel/plugin-transform-block-scoped-functions@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.523 | warning "@babel/preset-env > @babel/plugin-transform-block-scoping@7.18.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.523 | warning "@babel/preset-env > @babel/plugin-transform-classes@7.18.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.523 | warning "@babel/preset-env > @babel/plugin-transform-computed-properties@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.523 | warning "@babel/preset-env > @babel/plugin-transform-destructuring@7.18.0" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.523 | warning "@babel/preset-env > @babel/plugin-transform-dotall-regex@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-duplicate-keys@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-exponentiation-operator@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-for-of@7.18.1" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-function-name@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-literals@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.524 | warning "@babel/preset-env > @babel/plugin-transform-member-expression-literals@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-modules-amd@7.18.0" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-modules-commonjs@7.18.2" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-modules-systemjs@7.18.4" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-modules-umd@7.18.0" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-named-capturing-groups-regex@7.17.12" has unmet peer dependency "@babel/core@^7.0.0".
20:14:30.525 | warning "@babel/preset-env > @babel/plugin-transform-new-target@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-object-super@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-parameters@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-property-literals@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-regenerator@7.18.0" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-reserved-words@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.526 | warning "@babel/preset-env > @babel/plugin-transform-shorthand-properties@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.527 | warning "@babel/preset-env > @babel/plugin-transform-spread@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.527 | warning "@babel/preset-env > @babel/plugin-transform-sticky-regex@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.527 | warning "@babel/preset-env > @babel/plugin-transform-template-literals@7.18.2" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.527 | warning "@babel/preset-env > @babel/plugin-transform-typeof-symbol@7.17.12" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.527 | warning "@babel/preset-env > @babel/plugin-transform-unicode-escapes@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.528 | warning "@babel/preset-env > @babel/plugin-transform-unicode-regex@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.528 | warning "@babel/preset-env > @babel/preset-modules@0.1.5" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.528 | warning "@babel/preset-env > babel-plugin-polyfill-corejs2@0.3.1" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.528 | warning "@babel/preset-env > babel-plugin-polyfill-corejs3@0.5.2" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.528 | warning "@babel/preset-env > babel-plugin-polyfill-regenerator@0.3.1" has unmet peer dependency "@babel/core@^7.0.0-0".
20:14:30.529 | warning "@babel/preset-env > @babel/plugin-proposal-class-properties > @babel/helper-create-class-features-plugin@7.18.0" has unmet peer dependency "@babel/core@^7.0.0".
20:14:30.529 | warning "@babel/preset-env > @babel/plugin-proposal-unicode-property-regex > @babel/helper-create-regexp-features-plugin@7.17.12" has unmet peer dependency "@babel/core@^7.0.0".
20:14:30.529 | warning "@babel/preset-env > babel-plugin-polyfill-corejs2 > @babel/helper-define-polyfill-provider@0.3.1" has unmet peer dependency "@babel/core@^7.4.0-0".
20:14:30.529 | warning " > @vue/cli-plugin-eslint@4.5.17" has incorrect peer dependency "eslint@>= 1.6.0 < 7.0.0".
20:14:30.529 | warning "@vue/cli-plugin-eslint > eslint-loader@2.2.1" has incorrect peer dependency "eslint@>=1.6.0 <7.0.0".
20:14:30.535 | warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-import@^2.22.1".
20:14:30.535 | warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-jsx-a11y@^6.4.1".
20:14:30.535 | warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react@^7.21.5".
20:14:30.535 | warning " > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react-hooks@^4 \|\| ^3 \|\| ^2.3.0 \|\| ^1.7.0".
20:14:30.536 | warning "eslint-config-airbnb > eslint-config-airbnb-base@14.2.1" has unmet peer dependency "eslint-plugin-import@^2.22.1".
20:14:30.536 | warning " > progress-bar-webpack-plugin@2.1.0" has unmet peer dependency "webpack@^1.3.0 \|\| ^2 \|\| ^3 \|\| ^4 \|\| ^5".
20:14:30.536 | warning " > sass-loader@7.3.1" has unmet peer dependency "webpack@^3.0.0 \|\| ^4.0.0".
20:14:30.559 | warning Workspaces can only be enabled in private projects.
20:14:41.840 | [5/5] Building fresh packages...
20:14:42.834 | Done in 30.51s.
20:14:42.866 | Running "yarn run build"
20:14:43.024 | yarn run v1.22.17
20:14:43.054 | $ vue-cli-service build
20:14:43.884 | Browserslist: caniuse-lite is outdated. Please run:
20:14:43.885 | npx browserslist@latest --update-db
20:14:43.885 | Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
20:14:43.914 |  
20:14:43.915 | -  Building for production...
20:14:44.142 | WARN  A new version of sass-loader is available. Please upgrade for best experience.
20:14:46.316 | Browserslist: caniuse-lite is outdated. Please run:
20:14:46.317 | npx browserslist@latest --update-db
20:14:46.318 | Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
20:15:43.488 | WARNING  Compiled with 2 warnings7:15:43 PM
20:15:43.489 |  
20:15:43.490 | warning
20:15:43.490 |  
20:15:43.490 | asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
20:15:43.490 | This can impact web performance.
20:15:43.490 | Assets:
20:15:43.491 | fonts/Raleway-Variable.3b5635ac.ttf (302 KiB)
20:15:43.491 | js/chunk-vendors.46ca87ff.js (5.98 MiB)
20:15:43.491 | js/dashy.0fac8667.js (745 KiB)
20:15:43.491 |  
20:15:43.497 | warning
20:15:43.497 |  
20:15:43.497 | entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
20:15:43.497 | Entrypoints:
20:15:43.498 | dashy (6.94 MiB)
20:15:43.498 | css/chunk-vendors.d8067ad8.css
20:15:43.498 | js/chunk-vendors.46ca87ff.js
20:15:43.498 | css/dashy.f41729a5.css
20:15:43.498 | js/dashy.0fac8667.js

<!--EndFragment-->
</body>
</html>
```

**Comment by CrazyWolf13** [comment]
Other actions like recompiling throw the same error, I guess there is a problem with writing to the conf file?

**Comment by Lissy93** [comment]
Hey @vishalkadam7,
Sorry for the late response. If you've deployed to a static host, like Vercel, then there is no server-side code or Node server running, so actions which call a backend endpoint (saving config + rebuilding + server-side status checks) will not work unfortunately. This is documented a bit more in the [Deployment --> Cloud Service](https://github.com/Lissy93/dashy/blob/master/docs/deployment.md#deploy-to-cloud-service) docs.

The once exception is Netlify, as I've written cloud function endpoints for all the server-side endpoints, except for saving config and recompiling the app, which isn't possible to do.

Instead, after making changes, you'll need to copy the generated YAML and paste it into your conf.yml in your GitHub repo, then Vercel or Netlify will automatically rebuild and redeploy with your new changes. Sorry, I know that's not super ideal, but just a limitation of those services.

**Comment by vishalkadam7** [comment]
Thanks for the brief now I understand where I was wrong, I want to ask would it be possible to build dashy from the source code, I did try the docs provided by you but as a non developer I do understand few and I'll be even able to customise as required, but I'm stuck with the errors while building as there are couple of version conflicts one I solve the other comes in. Is there a particular branch that I should use for building for source code or it's the default branch

**Comment by CrazyWolf13** [comment]
Hi 
Yeah for sure you can build dashy from the source. Even though I don't fully understand why you build from source without developer knowledge.

Dependencies:

Node 16.0 LTS
git
yarn

https://github.com/Lissy93/dashy?tab=readme-ov-file#deploying-from-source-

You can simple clone the github master branch with
`git clone https://github.com/lissy93/dashy`

Cd into dashy folder:
`cd dashy`

install dependencies:
`yarn`

Build 
`yarn build`

Start
`yarn start`

**Comment by vishalkadam7** [comment]
actually I have a full-stack Next.js app with multi-tenancy with home page using this - template https://github.com/vercel/platforms
I'm trying to add dashy as a secondary workspace for my team

**Comment by vishalkadam7** [comment]
I'm trying to add dashy as a secondary workspace for my team and other individuals need very minimal options so I'll be customizing as required and disable or remove the once which are not required.

I have built a basic workspace using html, css and little js.
https://vops-workspace.vercel.app/
- iframe with Multi-tasking support.

now I'm trying to convert this into a full-stack Next.js app with multi-tenancy. So that anyone can add there fav bookmarks which would be visible to all the Artists (users), or can have an option to make it public or private. easy access to all your bookmarks within a single interface.
Instead of building it from scratch I thought of using dashy. So I'm here to explore dashy. need support

This is my first opensource project as a non Developer professionally I'm a 3D VFX Artist

I could use other dashboards for minimal use case but I have tried almost everything they do not support workspace view.
